Lib/ssl.py

```diff
@@ -1092,6 +1108,14 @@ def context(self, ctx):
         self._context = ctx
         self._sslobj.context = ctx
 
+    @property
+    def outer_server_hostname(self) -> str:
+        """The server name used in the outer ClientHello."""
+        if self._sslobj:
+            return self._sslobj.get_ech_status()[2]
+        else:
+            raise ValueError("No SSL wrapper around " + str(self))
```

Comment:
```suggestion
    def outer_server_hostname(self):
        """The server name used in the outer ClientHello."""
        if self._sslobj:
            return self._sslobj.get_ech_status()[2]
```

Other properties actually return None if sslobj is None.